### PR TITLE
Passing SDK config params to injected provider

### PR DESCRIPTION
**Pull request by fan-zhang-sv**
### _Summary_

extended the setAppInfo pattern to RN's in app browser provider. 
This is used to block EOA account to connect to smartWalletOnly app, and also enables adding data suffix in future RN smart wallet account

we are passing flattened params, including metadata and preference, this is for maximizing compatibility across App SDK version in case of future App SDK config structure change

### _How did you test your changes?_

Manually, demo attached in RN's PR

### Demo: noop on current extension
https://github.com/user-attachments/assets/debb56a9-22d3-4aef-9cc7-f7a08b14436e

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/util/provider.ts [hunk 1]
```diff
@@ -56,17 +56,19 @@ export function getCoinbaseInjectedProvider({
...
   preference,
 }: Readonly<ConstructorOptions>): ProviderInterface | undefined {
+  const { appName, appLogoUrl, appChainIds } = metadata;
+
   if (preference.options !== 'smartWalletOnly') {
     const extension = getCoinbaseInjectedLegacyProvider();
     if (extension) {
-      const { appName, appLogoUrl, appChainIds } = metadata;
-      extension.setAppInfo?.(appName, appLogoUrl, appChainIds);
+      extension.setAppInfo?.(appName, appLogoUrl, appChainIds, preference);
```
Comment (fan-zhang-sv) [inline packages/wallet-sdk/src/util/provider.ts:64]: adding it for extension too, just in case we need to use it, no harm even if we dont

Comment (cb-jake) [reply packages/wallet-sdk/src/util/provider.ts:64]: Did you test this on the extension side? Maybe include a video in the PR description?

**Review by cb-jake — APPROVED** [review]

File: packages/wallet-sdk/src/util/provider.ts (hunk 1 above)
Comment (fan-zhang-sv) [reply packages/wallet-sdk/src/util/provider.ts:64]: added demo!

**Pull request merged**